===================================================================
RCS file: /cvs/src/src/binutils/stabs.c,v
retrieving revision 1.34
retrieving revision 1.35
diff -u -r1.34 -r1.35
--- src/binutils/stabs.c	2012/03/30 12:51:07	1.34
+++ src/binutils/stabs.c	2013/01/15 13:41:05	1.35
@@ -2758,9 +2758,8 @@
              argtypes string is the mangled form of the argument
              types, and the full type and the physical name must be
              extracted from them.  */
-	  if (! stub)
-	    physname = argtypes;
-	  else
+	  physname = argtypes;
+	  if (stub)
 	    {
 	      debug_type class_type, return_type;
 
@@ -2879,9 +2878,7 @@
 		   || CONST_STRNEQ (argtypes, "__dt"));
   is_v3 = argtypes[0] == '_' && argtypes[1] == 'Z';
 
-  if (is_destructor || is_full_physname_constructor || is_v3)
-    *pphysname = argtypes;
-  else
+  if (!(is_destructor || is_full_physname_constructor || is_v3))
     {
       unsigned int len;
       const char *const_prefix;
